File: Hackerrank/WeekOfCode/WoC25/AppendAndDelete.py
"""
Week of Code 25 - https://www.hackerrank.com/contests/w25/challenges/append-and-delete/
"""
def numCommon(s,t):
    count = 0
    while count<min(len(s),len(t)) and s[count]==t[count]:
        count+=1
    return count

s = raw_input().strip()
t = raw_input().strip()
k = int(raw_input().strip())

common = numCommon(s,t)
nLeftS =  len(s)-common
nLeftT = len(t)- common
#print nLeftS,nLeftT

if (k>(len(s)+len(t))):
    print "Yes"
elif (k>=(nLeftS+nLeftT) and (k-nLeftS-nLeftT)%2==0):
    print "Yes"
else:
    print "No"

File: Hackerrank/WeekOfCode/WoC25/AppendAndDelete.cpp
/*
Week of Code 25 - https://www.hackerrank.com/contests/w25/challenges/append-and-delete/

*/

#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
#include <string>
using namespace std;

int countCommon(string s,string t ){
    int count = 0;
    int minl = min(s.size(),t.size());
    while(count<minl&&s[count]==t[count]){
        count+=1;
    }
    return count;
}

int main() {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */
    string s;
    string t;
    int k;
    cin>>s;
    cin>>t;
    cin>>k;
    if (k>=(s.size()+t.size())){
        cout<<"Yes"<<endl;
    }else if (k>=(s.size()-2*countCommon(s,t)+t.size()) && ((k-(s.size()-2*countCommon(s,t)+t.size()))%2==0)){
        cout<<"Yes"<<endl;
    }else{
        cout<<"No"<<endl;
    }
    return 0;
}
